.github/workflows/ci.yml

```
@@ -0,0 +1,152 @@
+
+env:
+  PYTHON_VERSION: "3.12"
+
+jobs:
+  # Pre-commit checks
+  pre-commit:
+    name: Pre-commit checks
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)